src/enclave/enclave_mem.c

```diff
@@ -98,98 +89,6 @@ static size_t addr_to_index(void* addr)
     return ((char*)mmap_end - (char*)addr) / PAGE_SIZE;
 }
 
-void* syscall_SYS_mmap(
-    void* addr,
-    size_t length,
-    int prot,
-    int flags,
-    int fd,
-    off_t offset)
-{
-    void* mem;
-    if ((flags & MAP_SHARED) && (flags & MAP_PRIVATE))
-    {
-        sgxlkl_warn("mmap() with MAP_SHARED and MAP_PRIVATE not supported\n");
-        errno = EINVAL;
-        mem = MAP_FAILED;
-    }
-    // Anonymous mapping/allocation
-    else if (fd == -1 && (flags & MAP_ANONYMOUS))
-    {
-        mem = enclave_mmap(addr, length, flags & MAP_FIXED, prot, 1);
-        if (mem == MAP_FAILED)
-            return mem;
-    }
-    // File-backed mapping (if allowed)
-    else if (fd >= 0 && enclave_mmap_flags_supported(flags, fd))
-    {
-        mem =
-            enclave_mmap(addr, length, flags & MAP_FIXED, prot | PROT_WRITE, 0);
-        if (mem > 0)
-        {
-            // Read file into memory
-            lseek(fd, offset, SEEK_SET);
```

Comment:
Did you intend to remove this lseek when this function was moved and partially rewritten?

Reply:
Yes.  lseek changes the offset in the fd, which is something mmap should not do.

Reply:
David replied elsewhere that it was intentional